plugins/inputs/prometheus/kubernetes.go

```diff
@@ -388,7 +354,7 @@ func registerPod(pod *corev1.Pod, p *Prometheus) {
 		p.lock.Lock()
 		defer p.lock.Unlock()
 	}
-	p.kubernetesPods[podURL.String()] = URLAndAddress{
+	p.kubernetesPods[PodID(pod.GetNamespace()+"/"+pod.GetName())] = URLAndAddress{
```

Comment:
Why prompted this change?

Reply:
Pod Informer gives us stream of Add/Update/Delete events, which we use to decide whether pod is of our interest or not. Natural key for the pod is "$namespace/$name" as it can't be changed for the duration of the pod life.

If we track pods by their natural keys, we avoid awkward conversion from pod spec to the URL every time just to check if we already track pod or not.